src/movies.js

```diff
@@ -1,16 +1,154 @@
 // Iteration 1: All directors? - Get the array of all directors.
+const getAllDirectors = movies => {
+    let getDirectors = movies.map(movie => movie.director)
+    return getDirectors
+    }
+
+console.log(getAllDirectors);
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+ 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+const howManyMovies = moviesArr => {
+       let filteredDrama = moviesArr.filter(function (movie) { return movie.genre.includes('Drama') && movie.director === 'Steven Spielberg' })
+       return filteredDrama.length
+};
+
+
+howManyMovies();
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+// const ratesAverage = movies.reduce((accum, currValue) => {
+//     return accum + currValue.rate
+// }, 0);
+
+// let avgMovies = ratesAverage / movies.length;
+// avgMovies = avgMovies.toFixed(2);
+
+function ratesAverage(movies) {
+    return movies.length ? Number((movies.reduce(function (acc, movie) {
+        return acc + (movie.rate ? movie.rate : 0);
+    }, 0) / movies.length).toFixed(2)) : 0;
+}
+
+ratesAverage();
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+    const dramaAveTotal = arr.filter(movies => movies.genre.includes('Drama'));
```

Comment:
After you got all the `dramaMovies` back you could just pass them as an argument to the `ratesAverage()` function previously defined. Your solution could be along these lines:
```
function dramaMoviesRate(someMovies) {
  let drMovies = someMovies.filter(eachMovie =>
    eachMovie.genre.includes('Drama')
  );
  return ratesAverage(drMovies);
}
```